azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -267,10 +269,16 @@ def _parse_agent_manifests(self, vm_settings):
         for family in families:
             name = family["name"]
             version = family.get("version")
+            is_version_from_rsm = family.get("isVersionFromRSM")
+            is_vm_enabled_for_rsm_upgrades = family.get("isVMEnabledForRSMUpgrades")
             uris = family.get("uris")
             if uris is None:
                 uris = []
             agent_family = VMAgentFamily(name, version)
+            if is_version_from_rsm is not None:
+                agent_family.is_version_from_rsm = is_version_from_rsm
```

Comment:
Related comment: naming a variable "is_*" creates a strong expectation that it is a bool, but these 2 variables seem to actually be tri_state: None, False, or True